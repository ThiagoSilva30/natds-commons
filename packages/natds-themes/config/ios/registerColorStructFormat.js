import fs from 'fs';
import path from 'path';
import Handlebars from 'handlebars';

const registerColorStructFormat = () => {
  const html = fs.readFileSync(path.resolve(__dirname, './templates/colorPaletteStruct.hbs')).toString();
  const htmlTemplate = Handlebars.compile(html);

  const formatter = (dictionary, platform) => htmlTemplate({
    color: dictionary.properties.color,
    structName: platform.files[0].destination.replace('.swift', ''),
  });

  return {
    formatter,
    name: 'ios-swift/struct-colors.swift',
  };
};

export default registerColorStructFormat;
