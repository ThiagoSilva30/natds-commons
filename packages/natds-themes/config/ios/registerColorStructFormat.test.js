import registerColorStructFormat from './registerColorStructFormat';

const dictionary = {
  properties: {
    color: {
      primary: {
        attributes: {},
        name: 'colorPrimary',
        original: {
          value: '#F091C9',
        },
        path: [
          'color',
          'primary',
        ],
        value: '#F091C9',
      },
    },
  },
};

const platform = {
  files: [
    {
      destination: 'AvonColorPaletteDark.swift',
    },
  ],
};

describe('registerColorPalletProtocolFormat', () => {
  it('should return the register config', () => {
    const register = registerColorStructFormat();

    expect(typeof register.formatter).toBe('function');
    expect(register.name).toBe('ios-swift/struct-colors.swift');
  });

  it('should return the correct formatter template', () => {
    const register = registerColorStructFormat();

    expect(register.formatter(dictionary, platform)).toMatchSnapshot();
  });
});
