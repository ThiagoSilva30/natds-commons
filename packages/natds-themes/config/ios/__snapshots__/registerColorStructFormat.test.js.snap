// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registerColorPalletProtocolFormat should return the correct formatter template 1`] = `
"struct AvonColorPaletteDark: ColorPalette {
  let primary = \\"#F091C9\\"
}"
`;
